### ISSUE TITLE
after install of catalina visual studio code can't find git

### ISSUE THREAD

**Issue by jserenson** [microsoft/vscode#82114]
I just installed the new catalina os on my iMac.
I have been using gitlens and git on Visual Studio Code successfully before the update
Since the install, It cannot find git.
Which I know is installed, and I installed the latest version just in case.
Now I cannot use git or gutless.

**Comment by andrecasal** [comment]
Good thing I just saw this issue before updating to Catalina, I literally had the update window opened 😁

**Comment by spahnke** [comment]
No problem here. I use the Homebrew version of Git (2.23.0) and the stable version of VS Code (1.38.1). GitLens is working as well for me.

**Comment by jserenson** [comment]
> No problem here. I use the Homebrew version of Git (2.23.0) and the stable version of VS Code (1.38.1). GitLens is working as well for me.

Hi I updated my post so it was clear that I am not able to use git or gitlens after installing the macosx Catalina update.

**Comment by jsejcksn** [comment]
@jserenson I am having the same issue after installing macOS 10.15. Why did you close this?

Needing to install Homebrew and install `git` via Homebrew is not an acceptable solution for me.

**Comment by spahnke** [comment]
@jserenson I know, I updated to Catalina yesterday and then saw your post. Just wanted to provide more information about a constellation where it does work 👍

**Comment by jserenson** [comment]
I have solved the problem.
I also was having a problem with Xcode 11.1 not updating.
After I moved Xcode to trash and reinstalled it for the app store.
I opened Xcode and it installed the components.
I then went into Xcode preferences source control and selected git tab
It showed my name and email
Then went to Locations, and selected Xcode 11.1 in the command line tools dropdown
I restarted my machine and started Anaconda and launched VSC and git was working

**Comment by andrecasal** [comment]
FYI, I updated to Catalina without any issues.

**Comment by jsejcksn** [comment]
@jserenson Thanks for sharing the solution. It also worked for me. The only difference for me was that I already had Xcode 11.1 installed, so I just had to open Preferences and select the cli tools version in the Locations tab. To be certain, I also closed Xcode, ran

```shell
xcode-select --install
```

and then reopened Xcode to check the setting I just updated. Nothing was changed and everything works accordingly now.

**Comment by DanWahlin** [comment]
I hit the issue as well today but the steps provided by @jserenson and @jsejcksn got it working. Thanks for sharing the info.

**Comment by bymayo** [comment]
FYI upgrading to VScode 1.39 fixes this issue

**Comment by MikeTodd** [comment]
I'm on 1.39.2 and still had this problem. The above solutions did work for me though. To recap:

* Move Xcode to trash.
* Go to App Store, reinstall Xcode.
* On the command line, run "xcode-select --install" to install command line tools.
* Note: During install, I was notified that an update was available for Xcode Command Line Tools. I waited for the install to finish, and then that update notification went away.
* Restarted VS Code.

**Comment by bymayo** [comment]
@MikeTodd I found that reinstalling Git for Mac also solved this issue on my MacBook Pro, the above fix only worked for me on my iMac (Must be different setups!)

**Comment by S-n-d** [comment]
@jsejcksn 's solution works perfectly, thanks! 
Don't forget to close and open xcode after install.

**Comment by armchairdeity** [comment]
Yeah, I started having this problem "out of the blue" (or so I thought) this morning, until I realized I had installed an Xcode update. Since I had just installed the update, I followed @jserenson's instructions... specifically checking Xcode prefs for names in the SCM panel and setting the location. Testing after that verified that the problem was resolved.

Thank you, @jserenson ! I'd probably still be farting around with it without you!

**Comment by sathishpaul** [comment]
Thanks for providing pointers and solutions to this problem.

I wanted to chime in and mention that I was able to solve the issue by updating Xcode (using the App Store) and also using the terminal `xcode-select --install` and allowing it to complete downloading the necessary tools. I did NOT have to move Xcode to trash. I restarted VS code (also updated to 1.40.1) and the problem went away.

Hope this helps, but your mileage may vary.
